File: internal/models/message.model.go
package models

import "time"

type Message struct {
	ID             int64     `json:"id"`
	Message        string    `json:"message"`
	SenderId       int64     `json:"senderId"`
	ConversationId int64     `json:"conversation"`
	Read           int64     `json:"read"`
	CreatedAt      time.Time `json:"createdAt"`
	Sender         *UserDTO  `json:"sender"`
}

File: web/src/app/core/models/message.model.ts
import {IUser} from "./user.model";

export interface IMessage {
    id?: number;
    message: string;
    createdAt?: string | Date;
    sender?: IUser;
    conversation?: number;
    senderId?: number;
}

export interface IMessages {
    date: string;
    messages: IMessage[];
}